### PR TITLE
fix: Fix trapezoid accelerated inside check

### PR DESCRIPTION
**Pull request by andiwand**
This bug flagged some points to be inside while they are not. I fixed this by checking against two rectangles instead of just one which is too big.

discovered in https://github.com/acts-project/acts/pull/2694

### CODE REVIEW

**Comment by paulgessinger** [comment]
trap_abs
trap_tol

Here are some MC tests from the current `main` bounds implementation. This looks correct to me? Am I missing something?

**Comment by andiwand** [comment]
@paulgessinger this test https://github.com/acts-project/acts/blob/main/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp#L203 failed after I changed the RNG in https://github.com/acts-project/acts/pull/2694

The reason is that we previously extended the inner rectangle in X and Y direction using tolerance without special treatment of the corners. Not sure what we expect at the corners but the behavior is different in the polygon check and in the quick bounds check

looking at your MC generated image the problem are the corners where we either just inflate the trapezoid or like in your picture we have rectangles overshooting the inflated trapezoid

**Comment by paulgessinger** [comment]
Hm, I think that's what we want. By convention, the tolerances are aligned with the local axes not with the slanted edge of the trap

File: Core/src/Surfaces/TrapezoidBounds.cpp [hunk 1]
```diff
@@ -32,18 +32,25 @@ bool Acts::TrapezoidBounds::inside(const Acts::Vector2& lposition,
     double tolX = bcheck.tolerance()[eBoundLoc0];
     double tolY = bcheck.tolerance()[eBoundLoc1];
 
-    if (std::abs(y) - hlY > tolY) {
```
[the remark this answers is not in the document]
Comment (andiwand) [reply Core/src/Surfaces/TrapezoidBounds.cpp:35]: true that looks like the easier condition

File: Core/src/Surfaces/TrapezoidBounds.cpp [hunk 2]
```diff
@@ -32,18 +32,25 @@ bool Acts::TrapezoidBounds::inside(const Acts::Vector2& lposition,
...
+    // check outside y range + tolerance
+    if (std::abs(y) - hlY > tolY) {
       return false;
     }
 
-    if (std::abs(x) - std::min(hlXnY, hlXpY) <= tolX) {
-      // inside x range
+    // check inside x range + tolerance, inside y range
+    if ((std::abs(x) - std::min(hlXnY, hlXpY) <= tolX) &&
+        (std::abs(y) - hlY <= 0)) {
```
[the remark this answers is not in the document]
Comment (andiwand) [reply Core/src/Surfaces/TrapezoidBounds.cpp:47]: right I can simplify this

**Comment by andiwand** [comment]
the polygon code will make the trapezoid bigger in x and y so there are no additional edges which seems "more correct" to me

**Comment by paulgessinger** [comment]
@andiwand Hm I see. I think the polygon code calculates the minimum distance first and then determines if they're inside tolerance in x and y.

Ultimately it's a choice. I think the current behavior is least surprising given "dumb" tolerances but that's just me.

Btw, with this PR included, the covered shape by the trap with tolerances seems to be exactly unchanged compared to before:

trap_fix1

**Comment by andiwand** [comment]
hmm I see. now I am confused. I don't think the numbers where small enough for a FP issue but let me check again

this is the error I see in https://github.com/acts-project/acts/pull/2694 (reverted the fix there)

```
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3459": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2921219286961447; y = -2.2747078076516623; index = 864; tol = 0.29999999999999999; 

*** 1 failure is detected in the test module "ActsUnitTestTrapezoidBounds"
```

**Comment by paulgessinger** [comment]
@andiwand Hm, the way the test is written (full polygon vs trap inside method) I think you're diagnosis should be correct, in that it's the ears of the trap that are just different in the two. 😵‍💫

**Comment by andiwand** [comment]
running on higher stats shows that this happens very rarely and only in one of the corners. happened 52 / 1000000

my code is basically delegating the corners to the polygon code this is why the edge case disappears. maybe there is a problem in the polygon code after all?

```
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3459": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2921219286961447; y = -2.2747078076516623; index = 864; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_99227": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2927764731479421; y = -2.2671555780395343; index = 24806; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_393170": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1894354545938617; y = -2.1788371905737849; index = 98292; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_496539": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2864731227239243; y = -2.2921108686601772; index = 124134; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_588671": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2936507056208422; y = -2.2965274816838361; index = 147167; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_606942": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1842893412891726; y = -2.1855486353168425; index = 151735; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_607235": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2875169013653274; y = -2.275979377018901; index = 151808; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_641407": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2598524192181877; y = -2.2928325712956936; index = 160351; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_652759": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2921676676132137; y = -2.2612554178795308; index = 163189; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_798858": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.1993815231893876; y = -2.1992631381251639; index = 199714; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_935815": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2946771713483498; y = -2.2617424246103539; index = 233953; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_986510": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.18131398344633; y = -2.1856603826137593; index = 246627; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1073322": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.1881016953942876; y = -2.1878458042266722; index = 268330; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1096571": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2918306145022509; y = -2.2912138872779408; index = 274142; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1427843": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2987057942700302; y = -2.2995728147669983; index = 356960; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1454767": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2776473232870149; y = -2.2783245139687445; index = 363691; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1588987": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2829040593612069; y = -2.2915137252390174; index = 397246; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1619594": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1906196503171635; y = -2.186132009401466; index = 404898; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1625434": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.1774758600865791; y = -2.1870071669599302; index = 406358; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1625505": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.0914589899184506; y = -2.0965701622005968; index = 406376; tol = 0.10000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1633721": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.0982970311826028; y = -2.0900512542639484; index = 408430; tol = 0.10000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1725381": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.0881139017996766; y = -2.0967909743752013; index = 431345; tol = 0.10000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1825955": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2917429083972056; y = -2.2647751894650749; index = 456488; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_1937571": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2924152163099896; y = -2.2830196086180368; index = 484392; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2009135": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2988912269192454; y = -2.2885997734200769; index = 502283; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2027327": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2932345697543406; y = -2.2770510189855937; index = 506831; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2111059": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2692909953809615; y = -2.2996914415541774; index = 527764; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2163506": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.196244679758955; y = -2.1855463303575418; index = 540876; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2185507": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2908118771077479; y = -2.2763916894441936; index = 546376; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2200707": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2532876252156573; y = -2.297614299070641; index = 550176; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2236859": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2863688595931304; y = -2.2899106952404824; index = 559214; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2341683": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2586053569535425; y = -2.2899011753091534; index = 585420; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2374035": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2901310608520138; y = -2.2794285129702629; index = 593508; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2430543": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2768448658771838; y = -2.2869472650939331; index = 607635; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2437771": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2741561473126266; y = -2.295312605866656; index = 609442; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2683614": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1731514945032977; y = -2.1943859490713251; index = 670903; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2833647": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2857777249284243; y = -2.2692953502363742; index = 708411; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2869238": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1828500207741603; y = -2.1911049381802048; index = 717309; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2873310": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1948452884486311; y = -2.1841880918584211; index = 718327; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2928703": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.280459244194752; y = -2.29308077389959; index = 732175; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_2967289": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.0885187341909042; y = -2.0974608221421391; index = 741822; tol = 0.10000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3063175": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2912906121764962; y = -2.2958932731436312; index = 765793; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3201831": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2840423201565923; y = -2.2830118920890574; index = 800457; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3203351": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2867411949306726; y = -2.2722863640610726; index = 800837; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3308186": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1926017665480515; y = -2.1771373525811737; index = 827046; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3319959": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.2895660545932888; y = -2.2736561115160412; index = 829989; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3347742": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.1652598112084149; y = -2.1972461894165214; index = 836935; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3437493": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.0913848884277542; y = -2.092973276888733; index = 859373; tol = 0.10000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3494034": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.1913449872750412; y = -2.1935099404096041; index = 873508; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3499747": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2941721323315978; y = -2.2708707698860486; index = 874936; tol = 0.29999999999999999; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3751014": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = -1.1734667029918722; y = -2.1938266798218802; index = 937753; tol = 0.20000000000000001; 
/home/andreas/cern/source/acts/acts/git2/Tests/UnitTests/Core/Surfaces/TrapezoidBoundsTests.cpp(204): error: in "Surfaces/TrapezoidInsideCheck/_3831943": check bc.isInside({x, y}, vertices) == trapezoidBoundsObject.inside({x, y}, bc) has failed [false != true]
Failure occurred in a following context:
    x = 1.2685211202433315; y = -2.2899130013468794; index = 957985; tol = 0.29999999999999999; 

*** 52 failures are detected in the test module "ActsUnitTestTrapezoidBounds"
```

**Comment by paulgessinger** [comment]
If it's only in one of the corners maybe it's just `<` vs `<=` or something like that?

**Comment by andiwand** [comment]
it seems to be a rare numerical problem with the polygon method. I moved not the seed in https://github.com/acts-project/acts/pull/2694 so it does not fail

not sure if it is worth trying to make the polygon method more numerical stable

closing for now
